core/aws-core/src/main/java/software/amazon/awssdk/awscore/internal/authcontext/AwsCredentialsAuthorizationStrategy.java

```diff
@@ -73,8 +75,15 @@ public Signer resolveSigner() {
      */
     @Override
     public void addCredentialsToExecutionAttributes(ExecutionAttributes executionAttributes) {
-        AwsCredentialsProvider credentialsProvider = resolveCredentialsProvider(request, defaultCredentialsProvider);
-        AwsCredentials credentials = resolveCredentials(credentialsProvider, metricCollector);
+        IdentityProvider<? extends AwsCredentialsIdentity> credentialsProvider =
+            resolveCredentialsProvider(request, defaultCredentialsProvider);
+        // TODO: Other option is casting:
+        //       AwsCredentials credentials = (AwsCredentials) resolveCredentials(credentialsProvider, metricCollector);
+        //       This may be unsafe if there is a sub-type of AwsCredentialsIdentity that is not AwsCredentials.
+        //       Like if customer creates an OdinAwsCredentialsIdentity which directly is-a AwsCredentialsIdentity.
+        AwsCredentials credentials = CredentialUtils.toCredentials(resolveCredentials(credentialsProvider, metricCollector));
+
+        // TODO: Should the signer be changed to use AwsCredentialsIdentity? Maybe with Signer SRA work, not now.
```

Comment:
The signer definitely should... but I think it's okay to keep it as a TODO here in this PR.

+1 to not downcasting. Hopefully when we refactor the signers, we won't have to do these conversions any more.